lms/djangoapps/support/static/support/jsx/entitlements/data/api/client.js

```diff
@@ -0,0 +1,53 @@
+import 'whatwg-fetch';
+import Cookies from 'js-cookie';
+
+import { entitlementList } from './endpoints';
+
+const headers = {
+  Accept: 'application/json',
+  'Content-Type': 'application/json',
+  'X-CSRFToken': Cookies.get('csrftoken'),
+}
+
+export function requestEntitlements({usernameOrEmail}) {
+  return fetch(
+    `${entitlementList}/${usernameOrEmail}`, {
+      credentials: 'same-origin',
+      method: 'get'
+    },
+  );
+}
+
+export function createEntitlement({courseUuid, user, mode, reason, comments}) {
+  return fetch(
+    `${entitlementList}/${user}`, {
+      credentials: 'same-origin',
+      method: 'post',
+      headers: headers,
+      body: JSON.stringify({
+        course_uuid: courseUuid,
+        user: user,
+        mode: mode,
+        reason: reason,
+        comments: comments
+      })
+    }
+  );
+}
+
+export function updateEntitlement({email, reason, entitlementUuid, comments}) {
+  //Email param may be removable when EntitlementSupportListView
```

Comment:
Can we add a ticket id to this that we are planning to do this work in?  It looks like this should be tracked as a Task or story and a TODO.

Reply:
We haven't decided yet if its something we want to do/adds any value

Reply:
Should I make a ticket we can talk about in grooming?

Reply:
Lets talk about it before grooming.  If we believe it is work we should do it is better to have it in a ticket than lost in the code simply in a comment.